protocol/x/clob/e2e/conditional_orders_test.go

```diff
@@ -958,6 +958,1125 @@ func TestConditionalOrder(t *testing.T) {
+				constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_SL_50250,
+			},
+			ordersForSecondBlock: []clobtypes.Order{
+				// Create a match with price $50,500.
+				// This price can trigger the conditional order if unbounded.
+				// The bounded price is $50,000 + $50,000 * 0.4% = $50,200, which would not trigger the conditional order.
+				constants.Order_Carl_Num1_Id0_Clob0_Buy1BTC_Price50500_GTB10,
+				constants.Order_Dave_Num1_Id0_Clob0_Sell1BTC_Price49997_GTB10,
+			},
+
+			expectedExistInState: map[clobtypes.OrderId]bool{
+				constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_SL_50005.OrderId: true,
+			},
+			expectedInTriggeredStateAfterBlock: map[uint32]map[clobtypes.OrderId]bool{
+				2: {constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_SL_50005.OrderId: false},
+				3: {constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_SL_50005.OrderId: false},
+				4: {constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_SL_50005.OrderId: false},
+			},
+		},
+		"TakeProfit/Sell conditional order is placed and not triggered by matched price": {
+			subaccounts: []satypes.Subaccount{
+				constants.Bob_Num0_100_000USD,
+				constants.Carl_Num1_100000USD,
+				constants.Dave_Num1_500000USD,
+			},
+			ordersForFirstBlock: []clobtypes.Order{
+				constants.ConditionalOrder_Bob_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_TP_50005,
+			},
+			ordersForSecondBlock: []clobtypes.Order{
+				// Create a match with price $50,003.
+				constants.Order_Carl_Num1_Id0_Clob0_Buy1BTC_Price50003_GTB10,
+				constants.Order_Dave_Num1_Id0_Clob0_Sell1BTC_Price49997_GTB10,
+			},
+
+			expectedExistInState: map[clobtypes.OrderId]bool{
+				constants.ConditionalOrder_Bob_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_TP_50005.OrderId: true,
+			},
+			expectedInTriggeredStateAfterBlock: map[uint32]map[clobtypes.OrderId]bool{
+				2: {constants.ConditionalOrder_Bob_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_TP_50005.OrderId: false},
+				3: {constants.ConditionalOrder_Bob_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_TP_50005.OrderId: false},
+				4: {constants.ConditionalOrder_Bob_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_TP_50005.OrderId: false},
+			},
+		},
+		"TakeProfit/Sell conditional order is placed and not triggered by matched price (bounded)": {
+			subaccounts: []satypes.Subaccount{
+				constants.Bob_Num0_100_000USD,
+				constants.Carl_Num1_100000USD,
+				constants.Dave_Num1_500000USD,
+			},
+			ordersForFirstBlock: []clobtypes.Order{
+				// Trigger price is $50,250.
+				constants.ConditionalOrder_Bob_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_TP_50250,
+			},
+			ordersForSecondBlock: []clobtypes.Order{
+				// Create a match with price $50,500.
+				// This price can trigger the conditional order if unbounded.
+				// The bounded price is $50,000 + $50,000 * 0.4% = $50,200, which would not trigger the conditional order.
```

Comment:
How did you get 0.4%? `const ConditionalOrderTriggerMultiplier uint64 = 5`, which means that `marketParam.MinPriceChangePpm` has to be 0.08% = 0.0008 = 800?

Reply:
ah good catch  - i meant to say 0.5% (0.1% * 5), but just double checked the test constants that we use, `min_price_change_ppm` = 50 😢 

created a new test constant (min_price_change_ppm = 1k) and updated tests and comments.